src/com/cloudogu/ces/cesbuildlib/Trivy.groovy

```diff
@@ -0,0 +1,133 @@
+                        break
+                    case TrivyScanStrategy.FAIL:
+                        script.archiveArtifacts artifacts: "${trivyReportFile}", allowEmptyArchive: true
+                        script.error("Trivy has found vulnerabilities in image " + imageName + ". See " + trivyReportFile)
+                        break
+                }
+                return false
+            default:
+                script.error("Error during trivy scan; exit code: " + exitCode)
+        }
+    }
+
+    /**
+     * Scans a dogu image for vulnerabilities.
+     * Notes:
+     * - Use a .trivyignore file for allowed CVEs
+     * - This function will generate a JSON formatted report file which can be converted to other formats via saveFormattedTrivyReport()
+     * - Evaluate via exit codes: 0 = no vulnerability; 1 = vulnerabilities found; other = function call failed
+     *
+     * @param doguDir The directory the dogu code (dogu.json) is located
+     * @param severityLevel The vulnerability level to scan. Can be a member of TrivySeverityLevel or a custom String (e.g. 'CRITICAL,LOW')
+     * @param strategy The strategy to follow after the scan. Should the build become unstable or failed? Or Should any vulnerability be ignored? (@see TrivyScanStrategy)
+     * @param additionalFlags Additional Trivy command flags
+     * @param trivyReportFile Location of Trivy report file. Should be set individually when scanning multiple images in the same pipeline
+     * @return Returns true if the scan was ok (no vulnerability found); returns false if any vulnerability was found
+     */
+    boolean scanDogu(
+        String doguDir = ".",
+        String severityLevel = TrivySeverityLevel.CRITICAL,
+        String strategy = TrivyScanStrategy.UNSTABLE,
+        // Avoid rate limits of default Trivy database source
+        String additionalFlags = "--db-repository public.ecr.aws/aquasecurity/trivy-db --java-db-repository public.ecr.aws/aquasecurity/trivy-java-db",
+        String trivyReportFile = "trivy/trivyReport.json"
+    ) {
+        String image = script.sh(script: "jq .Image ${doguDir}/dogu.json", returnStdout: true).trim()
+        String version = script.sh(script: "jq .Version ${doguDir}/dogu.json", returnStdout: true).trim()
+        return scanImage(image+":"+version, severityLevel, strategy, additionalFlags, trivyReportFile)
+    }
+
+    /**
+     * Save the Trivy scan results as a file with a specific format
+     *
+     * @param format The format of the output file (@see TrivyScanFormat)
+     * @param formattedTrivyReportFilename The file name your report files should get, without file extension. E.g. "ubuntu24report"
+     * @param trivyReportFile The "trivyReportFile" parameter you used in the "scanImage" function, if it was set
+     */
+    void saveFormattedTrivyReport(String format = TrivyScanFormat.HTML, String formattedTrivyReportFilename = "trivyReport", String trivyReportFile = "trivy/trivyReport.json") {
```

Comment:
My intuition was that the order of parameters should be: `trivyReportFile`, `format`, `formattedTrivyReportFilename`.

Also it feels strange to expect the user to leave out the file extension. This could also be a problem when we want to support custom formats as then we have no clue, which extension we should add via the code but it is not so easy to provide a default parameter with a matching filename. Maybe we could use the format parameter for the default parameter.

A third thing is, that there is no way at the moment to set the output path for the formattedReport
```suggestion
    void saveFormattedTrivyReport(String trivyReportFile = "trivy/trivyReport.json", String format = TrivyScanFormat.HTML, String formattedTrivyReportFilename = "$trivyDirectory/trivyReport.${format}") {
```

Reply:
Not sure if I agree with the file extension. Are custom formats really a thing? How would that work with the format? It doesn't hurt, if your suggestion works, but if not I would just leave it as is.

Reply:
If you want to use the standard report file, you can just call `saveFormattedTrivyReport(TrivyScanFormat.HTML)`. If we use your order, you always have to declare the (default) report file name, which you may not even know as a new user of this function:  `saveFormattedTrivyReport("trivy/trivyReport.json", TrivyScanFormat.HTML)`

Reply:
I don't think we need the user to be able to set the output path right now. It would lead to confusion, because if the output path is changed, the report files will not be automatically archived and therefore not be available at the Jenkins job.